core/src/wsv.rs

```diff
@@ -415,21 +429,27 @@ impl<W: WorldTrait> WorldStateView<W> {
     ///
     /// **Locking behaviour**: Holding references to blocks stored in the blockchain can induce
     /// deadlock. This limitation is imposed by the fact that blockchain is backed by [`dashmap::DashMap`]
-    #[inline]
-    pub fn blocks(&self) -> crate::block::ChainIterator {
-        self.blocks.iter()
+    // #[inline]
+    // pub fn blocks(&self) -> crate::block::ChainIterator {
+    // self.blocks.lock().iter()
+    // }
+
+    pub fn lock_read_blocks(&self) -> LockedChainViewer {
```

Comment:
I didn't think we'd be going for locking behavior, but @appetrosyan mentioned there may be some benefits.
The main question here is whether, if locking is a necessity, is it worth it in the sense that it offsets the price of cloning the elements out of a `DashMap`. Either in performance or memory usage

Reply:
I think it should be possible to do read access without locking using Rust rules... But it will work only if functions that actually mutates `WorldStateView` will accept `&mut self`. Why they doesn't do that actually?

Reply:
In performance as of right now, it isn't. I think the largest benefit will come from stability. If the network is more stable given the same load, performance will degrade less in a larger system where each fault requires a view change.